xmtp/src/conversation.rs

```diff
@@ -128,19 +137,33 @@ where
         Ok(())
     }
 
+    fn members(&self, conn: &mut DbConnection) -> Result<Vec<Contact>, ConversationError> {
+        let my_installations = self.client.my_other_devices(conn)?;
+        let peer_installations = self
+            .client
+            .get_contacts_from_db(conn, self.peer_address().as_str())?;
+
+        Ok(vec![my_installations, peer_installations].concat())
+    }
+
     pub async fn initialize(&self) -> Result<(), ConversationError> {
```

Comment:
Should we check the conversation state before doing everything else under initialize? And are we calling this every time `SecretConversation` is constructed, or at some other point?

Just wondering where this will fit into https://github.com/xmtp/libxmtp/blob/main/xmtp/README.md?plain=1#L85-L88 (modifications welcome if this is no longer accurate)

Reply:
That's a good call-out. I had to make some changes to `initialize` for compatibility, but I haven't really taken a proper look at what needs to change to make it compliant with the state machine spec.

One thing is definitely to check the state and make sure it hasn't already been initialized.